server/commands.lua

```diff
@@ -179,73 +165,132 @@ QBCore.Commands.Add('closeserver', Lang:t("command.closeserver.help"), {{ name =
-    local Player = QBCore.Functions.GetPlayer(tonumber(args[1]))
+lib.addCommand('setmoney', {
+    help = Lang:t("command.setmoney.help"),
+    params = {
+        { name = Lang:t("command.setmoney.params.id.name"), help = Lang:t("command.setmoney.params.id.help") },
+        { name = Lang:t("command.setmoney.params.moneytype.name"), help = Lang:t("command.setmoney.params.moneytype.help") },
+        { name = Lang:t("command.setmoney.params.amount.name"), help = Lang:t("command.setmoney.params.amount.help") }
+    },
+    restricted = "qbox.god"
+}, function(source, args)
+    local Player = QBCore.Functions.GetPlayer(tonumber(args[Lang:t("command.setmoney.params.id.name")]))
     if Player then
-        Player.Functions.SetMoney(tostring(args[2]), tonumber(args[3]))
+        Player.Functions.SetMoney(tostring(args[Lang:t("command.setmoney.params.moneytype.name")]), tonumber(args[Lang:t("command.setmoney.params.amount.name")]))
     else
         TriggerClientEvent('QBCore:Notify', source, Lang:t('error.not_online'), 'error')
     end
-end, 'admin')
+end)
 
 -- Job
-
-QBCore.Commands.Add('job', Lang:t("command.job.help"), {}, false, function(source)
+lib.addCommand('job', {
+    help = Lang:t("command.job.help"),
+    restricted = "qbox.user"
+}, function(source, _)
     local PlayerJob = QBCore.Functions.GetPlayer(source).PlayerData.job
     TriggerClientEvent('QBCore:Notify', source, Lang:t('info.job_info', {value = PlayerJob.label, value2 = PlayerJob.grade.name, value3 = PlayerJob.onduty}))
-end, 'user')
+end)
 
-QBCore.Commands.Add('setjob', Lang:t("command.setjob.help"), { { name = Lang:t("command.setjob.params.id.name"), help = Lang:t("command.setjob.params.id.help") }, { name = Lang:t("command.setjob.params.job.name"), help = Lang:t("command.setjob.params.job.help") }, { name = Lang:t("command.setjob.params.grade.name"), help = Lang:t("command.setjob.params.grade.help") } }, true, function(source, args)
-    local Player = QBCore.Functions.GetPlayer(tonumber(args[1]))
+lib.addCommand('setjob', {
+    help = Lang:t("command.setjob.help"),
+    params = {
+        { name = Lang:t("command.setjob.params.id.name"), help = Lang:t("command.setjob.params.id.help") },
+        { name = Lang:t("command.setjob.params.job.name"), help = Lang:t("command.setjob.params.job.help") },
+        { name = Lang:t("command.setjob.params.grade.name"), help = Lang:t("command.setjob.params.grade.help"), optional = true }
+    },
+    restricted = "qbox.god"
+}, function(source, args)
+    local Player = QBCore.Functions.GetPlayer(tonumber(args[Lang:t("command.setjob.params.id.name")]))
     if Player then
-        Player.Functions.SetJob(tostring(args[2]), tonumber(args[3]))
+        if args[Lang:t("command.setjob.params.grade.name")] then
+            Player.Functions.SetJob(tostring(args[Lang:t("command.setjob.params.job.name")]), tonumber(args[Lang:t("command.setjob.params.grade.name")]))
```

Comment:
Isn't args[Lang:t("command.setjob.params.job.name")] already guaranteed to be a string?

Reply:
Can you explain this logic? This used to be args[2] and args[3], but it's args of a locale?

Reply:
ox_lib doesn't use args as indexes but as the name you put in the params